src/test/java/com/commercetools/sync/commons/utils/SyncUtilsTest.java

```diff
@@ -117,36 +119,63 @@ void batchElements_WithNegativeSize_ShouldReturnNoBatches() {
     }
 
     @Test
-    void replaceReferenceIdWithKey_WithNullReference_ShouldReturnNullReference() {
-        final Reference<Object> keyReplacedReference = replaceReferenceIdWithKey(null,
+    void getReferenceWithKeyReplaced_WithNullReference_ShouldReturnNullReference() {
+        final Reference<Object> keyReplacedReference = getReferenceWithKeyReplaced(null,
             () -> Reference.of(Category.referenceTypeId(), "id"));
         assertThat(keyReplacedReference).isNull();
     }
 
     @Test
-    void replaceReferenceIdWithKey_WithExpandedCategoryReference_ShouldReturnCategoryReferenceWithKey() {
+    void getReferenceWithKeyReplaced_WithExpandedCategoryReference_ShouldReturnCategoryReferenceWithKey() {
         final String categoryKey = "categoryKey";
         final Category mockCategory = mock(Category.class);
         when(mockCategory.getKey()).thenReturn(categoryKey);
 
         final Reference<Category> categoryReference = Reference.ofResourceTypeIdAndObj(Category.referenceTypeId(),
             mockCategory);
 
-        final Reference<Category> keyReplacedReference = replaceReferenceIdWithKey(categoryReference,
+        final Reference<Category> keyReplacedReference = getReferenceWithKeyReplaced(categoryReference,
             () -> Category.referenceOfId(categoryReference.getObj().getKey()));
         assertThat(keyReplacedReference).isNotNull();
         assertThat(keyReplacedReference.getId()).isEqualTo(categoryKey);
     }
 
     @Test
-    void replaceReferenceIdWithKey_WithNonExpandedCategoryReference_ShouldReturnReferenceWithoutReplacedKey() {
+    void getResourceIdentifierWithKeyReplaced_WithExpandedReference_ShouldReturnCategoryResourceIdentifierWithKey() {
+        final String categoryKey = "categoryKey";
+        final Category mockCategory = mock(Category.class);
+        when(mockCategory.getKey()).thenReturn(categoryKey);
+
+        final Reference<Category> categoryReference = Reference.ofResourceTypeIdAndObj(Category.referenceTypeId(),
+            mockCategory);
+
+        final ResourceIdentifier<Category> keyReplacedReference = getResourceIdentifierWithKeyReplaced(categoryReference,
+            () -> ResourceIdentifier.ofId(categoryReference.getObj().getKey()));
+        assertThat(keyReplacedReference).isNotNull();
+        assertThat(keyReplacedReference.getId()).isEqualTo(categoryKey);
+    }
+
+    @Test
+    void getReferenceWithKeyReplaced_WithNonExpandedCategoryReference_ShouldReturnReferenceWithoutReplacedKey() {
         final String categoryUuid = UUID.randomUUID().toString();
         final Reference<Category> categoryReference = Reference.ofResourceTypeIdAndId(Category.referenceTypeId(),
             categoryUuid);
 
-        final Reference<Category> keyReplacedReference = replaceReferenceIdWithKey(categoryReference,
+        final ResourceIdentifier<Category> keyReplacedReference = getReferenceWithKeyReplaced(categoryReference,
```

Comment:
why not Reference, because we are testing `getReferenceWithKeyReplaced`

Reply:
nice catch: 41e6358